tables/tests/test_expression.py

```diff
@@ -265,9 +265,12 @@ def test00b_simple_scalars(self):
         if common.verbose:
             print("Computed expression:", repr(r1), r1.dtype)
             print("Should look like:", repr(r2), r2.dtype)
-        self.assertTrue(
-            r1.shape == r2.shape and r1.dtype == r2.dtype and r1 == r2,
-            "Evaluate is returning a wrong value.")
+        msg = f"Evaluate is returning a wrong value: {expr_str}\n{r1=}\n{r2=}"
+        self.assertEqual(r1.shape, r2.shape, msg=msg)
+        # In something like 2 * np.in16(3) + np.int16(2) the result is still a
+        # np.int16 in NumPy 2.0, so we shouldn't actually check this:
```

Comment:
It is not really clear to me why "we shouldn't actually check this".
Apparently the logic of the test is that the result of the `tb.Expr` evaluation should be totally equivalent to the result of pure Python `eval`.
In principle breaking this logic could could break something on the user side.

On the other hand the current `tb.Expr` behaviour is aligned to the `numexpr` one at the moment.

Honestly I'm not sure how to proceed here.
@ivilata , @FrancescAlted , any suggestion?

Reply:
Yeah NumPy changed it's up casting behavior (I think,?) so it's possible something needs to be updated.

Reply:
(Mind you, `f` in the test is `int(3)`, not `np.int16(3)`.)

So PyTables stays with the NumExpr type `int64`, but NumPy changed from `int64` to `int16`. Then relaxing the test seems ok to me, but we should at least check that the dtype kind stays consistent, i.e. `r1.dtype.kind == r2.dtype.kind`.

Reply:
Thanks @ivilata, your suggestion sounds a good compromise to me.
@larsoner if it is OK for you we could proceed in that way.